R/tax_unique.R

```diff
@@ -0,0 +1,279 @@
+#' Filter occurrences to unique taxa
+#'
+#' A function to filter a list of taxonomic occurrences to unique taxa of a
+#' predefined resolution. In many cases palaeobiologists achieve this by
+#' retaining only unique occurrences identified to the given taxonomic
+#' resolution, however here we also retain occurrences identified to a coarser
+#' resolution which are not already represented within the dataset.
+#' If we take the following set of occurrences:
```

Comment:
I worry some users might get confused by the difference between taxa and occurrences throughout. I'm not exactly sure how to fix that (and maybe I'm wrong), and I suppose this function could be used on occurrence dataframes or strictly taxonomic dataframes, which makes it perhaps more confusing (but also more useful). 

One example is here where you list a bunch of taxa but refer to them as occurrences. Presumably, there are occurrences associated with these taxa, but here without any occurrence information it just seems like a list of taxa.

Reply:
I guess the key difference here would be repeats of the same taxon names to signify multiple occurrences in the examples. I can elaborate in the documentation, but providing a full example for both scenarios could be quite unwieldy? The intention is that the function can be used for both, but there would just be more filtering involved for occurrences than for taxon lists.

Reply:
I've added explanation that the occdf can be occurrences or taxa in the documentation, but let me know if it needs more.